pnp/PnpHelper.cs

```diff
@@ -0,0 +1,192 @@
+                    $"}}";
+        }
+
+        /// <summary>
+        /// Helper to retrieve the required property value from the <see cref="TwinCollection"/> property update patch received from service-initiated update.
+        /// </summary>
+        /// <typeparam name="T">The data type of the required property, as defined in the DTDL interface.</typeparam>
+        /// <param name="collection">The <see cref="TwinCollection"/> property update patch received as a result of service-initiated update.</param>
+        /// <param name="propertyName">The property name, as defined in the DTDL interface.</param>
+        /// <param name="componentName">(optional) The name of the component in which the property is defined. Can be null for property defined under the root interface.</param>
+        /// <returns>A tuple indicating if the <see cref="TwinCollection"/> property update patch contains the required property update, and the corresponding property value.</returns>
+        public static (bool, T) GetPropertyFromTwin<T>(TwinCollection collection, string propertyName, string componentName = null)
+        {
+            // If the desired property update is for a root component or nested component, verify that property patch received contains the desired property update.
+            if (string.IsNullOrWhiteSpace(componentName))
+            {
+                return collection.Contains(propertyName) ? (true, (T)collection[propertyName]) : (false, default);
+            }
+            
+            if (collection.Contains(componentName))
+            {
+                JObject componentProperty = collection[componentName];
+                {
+                    return componentProperty.ContainsKey(propertyName) ? (true, componentProperty.Value<T>(propertyName)) : (false, default);
+                }
+            }
+
+            return (false, default);
+        }
+
+        /// <summary>
+        /// Create a plug and play compatible command name.
+        /// </summary>
+        /// <param name="commandName">The name of the command to be invoked.</param>
+        /// <param name="componentName">(optional) The name of the component from which the command is to be invoked. Can be null for command defined under the root interface.</param>
+        /// <returns>A plug and play compatible command name.</returns>
+        public static string CreatePnpCommandName(string commandName, string componentName = default)
+        {
+            return string.IsNullOrWhiteSpace(componentName) ? commandName : $"{componentName}*{commandName}";
+        }
+
+        /// <summary>
+        /// Create a plug and play compatible command payload.
+        /// </summary>
+        /// <param name="serializedPayload">The serialized payload, in the format defined in the DTDL interface.</param>
+        /// <returns>A plug and play compatible command payload.</returns>
+        public static string CreatePnpCommandRequestPayload(string serializedPayload)
+        {
```

Comment:
Command request? Does this mean this will be used on Service Sample as well?

Reply:
Yes, since I was using these helpers to test my sample, I added them into `PnpHelper` directly. 
I've informed @bikamani that she can use/ modify these helpers based on her findings.

Reply:
👍